fix: report error on Sentinel connection refused (#445)

SamBergeron opened the pull request:
This is a quick fix to report on a sentinel cluster not being able to connect. 
This will also report retries in `redis.on('end', ...` for subsequent retries of the Sentinel cluster, this behavior is similar to what happens on a failed connection with a single redis instance.
This is the quickest fix I could think of.

luin commented on lib/connectors/sentinel_connector.js:
```diff
@@ -48,21 +48,26 @@ SentinelConnector.prototype.connect = function (callback) {
+      if (lastError) {
+        error += ' Last error: ' + lastError.message;
+      }
+      return callback(new Error(error));
```
We shouldn't `callback()` with an error here when we still keep retrying since a callback must not be called more than once.

Instead, what about just emit an event (like "sentinelError") when all sentinels are unreachable just like when a connection to a standalone Redis server is down.

SamBergeron commented:
Any idea why `1) should raise error when all sentinel are unreachable and retry is disabled` is failing?
I added a second callback to call the `silentEmit` function from redis.js. Mainly because with "normal" connections you callback an error everytime and loop inside redis.js, but with sentinels it's a recursive method. What do you think of this? Thanks.

SamBergeron commented:
Added a callback to `silentEmit` so that if none of the sentinels can connect we report an error, also added `sentinelError` reports to know which of the individual sentinels aren't connection if you're listening for that.

SamBergeron commented:
Just following up on this? Thoughts @luin ?

luin commented:
Sorry for the delay. Merged! 🍻
